common/djangoapps/philu_commands/management/commands/auto_generate_certificates_for_open_courses.py

```diff
@@ -29,8 +29,19 @@
 GeneratedCertificate = apps.get_model('certificates', 'GeneratedCertificate')
 
 
+def has_active_certificate(course):
+    has_activated_certificate = False
+    certificates = course.certificates
+
+    if certificates:
+        has_activated_certificate = any([certificate['is_active'] for certificate in certificates['certificates']])
+
+    return has_activated_certificate
+
+
 def is_course_valid_for_certificate_auto_generation(course):
-    return bool(course.has_started() and not course.has_ended() and course.may_certify())
+    return bool(
+        course.has_started() and not course.has_ended() and course.may_certify()) and not has_active_certificate(course)
```

Comment:
@qasimgulzar please test it, We have to remove this `not` with the new condition we have introduced in this method. `and has_active_certificate(course)` instead of `and not has_active_certificate(course)`